Src/Newtonsoft.Json/Utilities/EnumUtils.cs

```diff
@@ -247,6 +247,14 @@ public static object ParseEnumName(string enumText, bool isNullable, bool disall
                 }
             }
 
+            // Try parse case sensitive
+            try
+            {
+                return Enum.Parse(t, finalEnumText, false);
+            }
+            catch(ArgumentException) { }
```

Comment:
This will completely kill performance.

Reply:
I have now changed this.
The method that i have chosen is using more memory and a extra initilization time instead of each time a enum is deserilized. See EnumBidirectionalDictionary